File: algorithms/haskell/FibonacciRecursive.hs
import Data.Time.Clock
import System.IO
import Control.Exception (evaluate)

modVal :: Integer
modVal = 1000000007

fib :: Int -> Integer
fib 0 = 0
fib 1 = 1
fib n = (fib (n-1) + fib (n-2)) `mod` modVal

main :: IO ()
main = do
    hSetBuffering stdout LineBuffering
    contents <- getContents
    let n = read (head (words contents)) :: Int

    start  <- getCurrentTime
    result <- evaluate (fib n)
    end    <- getCurrentTime

    let ms = realToFrac (diffUTCTime end start) * 1000 :: Double
    print result
    hPutStrLn stderr $ show ms

File: algorithms/lua/fibonacci-recursive.lua
local MOD = 1000000007

local function fib(n)
    if n <= 1 then return n end
    return (fib(n-1) + fib(n-2)) % MOD
end

local function main()
    local n     = tonumber(io.read("*l"))
    local start = os.clock()
    local result = fib(n)
    local ms    = (os.clock() - start) * 1000

    print(result)
    io.stderr:write(string.format("%.3f\n", ms))
end

main()